acc_acc_modules/1_annotate_acc_modules.py

```diff
@@ -0,0 +1,373 @@
+
+# # Add annotations
+#
+# This notebook takes the dataframe with information about module composition and their labels and adds additional annotations including:
+#
+# 1. Which gene is contained within the modules (both gene id and gene name)
+# 2. Baseline expression and expression in some context of interest
+# 3. How clustered the module is on the genome
+# 4. KEGG pathways that genes are found in
+# 5. GO pathways genes are found in
+# 6. Regulon/operon genes are found in
+#
+# All this information will help _P. aeruginosa_ experiments filter and determine which module might be interesting to explore.
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import random
+import scipy
+import pandas as pd
+import numpy as np
+from itertools import product
+from core_acc_modules import paths
+
+random.seed(1)
+# -
+
+# User param
+method = "affinity"
+
+# +
+# Import gene memberships
+pao1_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pao1_modules_{method}_acc.tsv"
+)
+pa14_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pa14_modules_{method}_acc.tsv"
+)
+
+pao1_membership = pd.read_csv(pao1_membership_filename, sep="\t", index_col=0, header=0)
+pa14_membership = pd.read_csv(pa14_membership_filename, sep="\t", index_col=0, header=0)
+
+# +
+# Import gene metadata
+pao1_gene_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_gene_annot_filename = paths.GENE_PA14_ANNOT
+
+pao1_gene_annot = pd.read_csv(pao1_gene_annot_filename, index_col=0, header=0)
+pa14_gene_annot = pd.read_csv(pa14_gene_annot_filename, index_col=0, header=0)
+# -
+
+# Import metadata of samples
+metadata_filename = paths.SAMPLE_METADATA
+
+pao1_gene_annot = pao1_gene_annot["Name"].to_frame("gene name")
+pa14_gene_annot = pa14_gene_annot["Name"].to_frame("gene name")
+
+# ## Add gene names
+
+# Add gene names
+pao1_gene_module_labels = pao1_membership.merge(
+    pao1_gene_annot, left_index=True, right_index=True
+)
+pa14_gene_module_labels = pa14_membership.merge(
+    pa14_gene_annot, left_index=True, right_index=True
+)
+
+print(pao1_gene_module_labels.shape)
+pao1_gene_module_labels.head()
+
+print(pa14_gene_module_labels.shape)
+pa14_gene_module_labels.head()
+
+# ## Add expression information
+#
+# 1. What is the baseline level of expression for each gene in the module?
+# 2. What is the expression level of genes in a clinical context (i.e. clinical samples)?
+
+# Read in expression data
+pao1_compendium = pd.read_csv(paths.PAO1_COMPENDIUM, sep="\t", index_col=0)
+pa14_compendium = pd.read_csv(paths.PA14_COMPENDIUM, sep="\t", index_col=0)
+
+pao1_compendium.head()
+
+# Calculate median expression across all samples
+pao1_median_all = pao1_compendium.median().to_frame("median expression")
+pa14_median_all = pa14_compendium.median().to_frame("median expression")
+
+pao1_median_all.head()
+
+
+# Select subset of samples and calculate the median expression across that subset of samples
+# TO DO: Move this into utils
```

Comment:
Do you want to do this now?

Reply:
Nice catch! I forgot I wanted to do this. I'll update this so the notebook is cleaner